### PR TITLE
Follow-up to admin API to re-activate accounts

### LINKED ISSUE #7220: Reactivate an account
### Description

Deactivating an account works but trying to reactivate an account does not.
As far as I can tell from synapse code there is no handler for account reactivation.
https://github.com/matrix-org/synapse/blob/master/synapse/rest/admin/users.py#L229
If this is not implemented, wouldn't it be good to point it out in docs? https://github.com/matrix-org/synapse/blob/master/docs/admin_api/user_admin_api.rst

### Steps to reproduce

- deactivate an account using rest api
- account is properly deactivated
- try and reactivate the account

### Version information

- **Homeserver**: 1.12.3+bionic1

- **Install method**: apt

- **Platform**: ubuntu 18.04

### PR DESCRIPTION
**Pull request by clokep**
A follow-up to #7847 (which fixed #7220) to fix a couple of bugs, now the re-activated user is:

* Re-added to the user directory.
* Removed from the `erased_users` table.

### CODE REVIEW

**Comment by clokep** [comment]
CC @dklimpel -- hopefully this meets all your feedback!

**Comment by dklimpel** [comment]
> CC @dklimpel -- hopefully this meets all your feedback!

Yes, it does.

File: synapse/handlers/deactivate_account.py [hunk 1]
```diff
@@ -222,13 +223,24 @@ async def activate_account(self, user_id: str) -> None:
         """
         Activate an account that was previously deactivated.
 
-        This simply marks the user as activate in the database and does not
+        This marks the user as activate in the database, but does not
         attempt to rejoin rooms, re-add threepids, etc.
```
[the remark this answers is not in the document]
Comment (clokep) [reply synapse/handlers/deactivate_account.py:230]: Thanks! I updated this comment a bit in d06fd486f43f144af029edc6bcf991dadfb5006a, let me know if it makes more sense now?

[the remark this answers is not in the document]
Comment (anoadragon453) [reply synapse/handlers/deactivate_account.py:230]: Sounds good!

**Comment by JOJ0** [comment]
Hi, sorry to hijack this thread. How would I use this feature to re-activate an account? What is the API endpoint to send the request to? It's pretty urgent as I accidentally deactivated my main-account on my homeserver. I still have another admin account to send to the API.

I tried curls against endpoint addresses like this but no luck so far. Couldn't find it in the code either....didn't search extensively though....

https://my-hs.fqdn/_matrix/client/r0/admin/activate/%40${USER}%3Amy-hs.fqdn
at?access_token=${TOKEN}"

**Comment by clokep** [comment]
Please ask support questions in #synapse:matrix.org, but you might want to check the docs first: https://github.com/matrix-org/synapse/blob/master/docs/admin_api/user_admin_api.rst#create-or-modify-account

**Comment by localguru** [comment]
@clokep @anoadragon453 Hi, I reactivated a User using
```
curl --header "Authorization: Bearer ${ACCESS_TOKEN}" -XPUT -d \
    '{"deactivated": false, "password": "test"}' \
    http://localhost:8008/_synapse/admin/v2/users/${MATRIX_ID}
```
but the user is still in erased_table. After an hour the entry was gone. Didn't restart the server.

**Comment by mloskot** [comment]
> Please ask support questions in #synapse:matrix.org, but you might want to check the docs first: https://github.com/matrix-org/synapse/blob/master/docs/admin_api/user_admin_api.rst#create-or-modify-account


FYI @JOJ0 & @clokep, this is not possible. I've issued request to the support team and here is what I have received:

> Hi Mateusz
>
> Unfortunately it is impossible to verify the identity of deleted users. Per the matrix.org Privacy Notice ( https://matrix.org/legal/privacy-notice#2211-account-and-profile-information ): "Your username is stored indefinitely to avoid account recycling."
>
>To protect the security and privacy of our users we never reactivate or release deleted usernames.
>We recommend you create a new account using a different Matrix ID.
>
>Best wishes
>Matrix.org Support

Bummer!

**Pull request merged**